src/dotnet/commands/dotnet-add-p2p/Program.cs

```diff
@@ -0,0 +1,174 @@
+            {
+                Name = "dotnet add p2p",
+                FullName = ".NET Add Project to Project (p2p) reference Command",
+                Description = "Command to add project to project (p2p) reference",
+                AllowArgumentSeparator = true,
+                ArgumentSeparatorHelpText = "Project to project references to add"
+            };
+
+            app.HelpOption("-h|--help");
+
+            CommandArgument projectArgument = app.Argument("<PROJECT>",
+                "The project file to modify. If a project file is not specified," +
+                " it searches the current working directory for an MSBuild file that has a file extension that ends in `proj` and uses that file.");
+
+            CommandOption frameworkOption = app.Option("-f|--framework <FRAMEWORK>", "Add reference only when targetting a specific framework", CommandOptionType.SingleValue);
+
+            app.OnExecute(() => {
+                if (projectArgument.Value == null)
+                {
+                    throw new GracefulException("Argument <Project> is required.");
+                }
+
+                ProjectRootElement project = File.Exists(projectArgument.Value) ?
+                                                GetProjectFromFileOrThrow(projectArgument.Value) :
+                                                GetProjectFromDirectoryOrThrow(projectArgument.Value);
+
+                if (app.RemainingArguments.Count == 0)
+                {
+                    throw new GracefulException("You must specify at least one reference to add.");
+                }
+
+                int numberOfAddedReferences = AddProjectToProjectReference(project, frameworkOption.Value(), app.RemainingArguments);
+
+                if (numberOfAddedReferences != 0)
+                {
+                    project.Save();
+                }
+
+                return 0;
+            });
+
+            try
+            {
+                return app.Execute(args);
+            }
+            catch (GracefulException e)
+            {
+                Reporter.Error.WriteLine(e.Message.Red());
+                app.ShowHelp();
+                return 1;
+            }
+        }
+
+        // There is ProjectRootElement.TryOpen but it does not work as expected
+        // I.e. it returns null for some valid projects
+        public static ProjectRootElement TryOpenProject(string filename)
+        {
+            try
+            {
+                return ProjectRootElement.Open(filename, new ProjectCollection(), preserveFormatting: true);
+            }
+            catch (Microsoft.Build.Exceptions.InvalidProjectFileException)
+            {
+                return null;
+            }
+        }
+
+        public static ProjectRootElement GetProjectFromFileOrThrow(string filename)
+        {
+            if (!File.Exists(filename))
+            {
+                throw new GracefulException($"Provided project `{filename}` does not exist.");
```

Comment:
How about Project `{}`... does not exist? I have not seen us use this provided before. Did you get this from @blackdwarf? If so, that's ok, this is his domain. If not, check with him.